pretyping/detyping.ml

```diff
@@ -883,7 +883,9 @@ and detype_binder d flags bk avoid env sigma decl c =
   | BLetIn ->
       let c = detype d { flags with flg_isgoal = false } avoid env sigma (Option.get body) in
       (* Heuristic: we display the type if in Prop *)
-      let s = try Retyping.get_sort_family_of (snd env) sigma ty with _ when !Flags.in_debugger || !Flags.in_toplevel -> InType (* Can fail because of sigma missing in debugger *) in
+      let s =
+        try Retyping.get_sort_family_of (snd env) sigma ty
+        with Retyping.RetypeError _ -> InType (* eg: the type is an evar *) in
```

Comment:
```suggestion
        with Retyping.RetypeError _ -> InType (* eg: the type is an evar *) 
      in
```

Reply:
I'm not a big fan of this style, a line with just a `in` does not seem worth it. Is there a reason for it?

Reply:
It's much easier to find the end of the letin that way imo

Reply:
Well, indentation helps as well no?

Some styles help patches to be smaller, changes more local... I'm all for that.
This one looks a bit too much about taste to me (even re-running CI looks too much compared to what we could gain, imo).

Reply:
@gares: please keep the information that it can fail also in the debugger or in the OCaml toplevel.

Reply:
I thought about that, but then I thought it was outdated, from a time where sigma was not passed explicitly. Now that we pass it, what is the difference when we run it from the debugger? If I'm wrong I'll add it back, sure.

Reply:
The debugging printers run in the ocamldebug process and are not aware of the `Global.env` of the process they are debugging. That would be nice if ocamldebug were able to pass other information to the printers than the term to print but I'm not sure it is possible. To be honest I did not check if it is possible but as it is now, it is certainly not the case. As another example, the nametab is unused by the debugging printers.

Reply:
ok, I'll restore the comment then, and while I'm at it make @SkySkimmer happy